### ISSUE TITLE
Grid Consumption and  Return to grid

### ISSUE THREAD

**Issue by superoloyt** [macxq/foxess-ha#188]
I have a problem with adding the Grid consumption and return to grid to HA. I attached a screenshot with error.
Thank you in advance.

![image](https://github.com/macxq/foxess-ha/assets/69038702/0af2ffa5-1362-44b5-9a36-4dafffd72465)

**HA version:**
2023.7.3

**foxess-ha version:**
v0.31

**Comment by superoloyt** [comment]
> use FoxESS - Energy Generated for solar production
The dolar production is displayed. The problem are the errors in the orange.

**Comment by Kepro** [comment]
yes, because you're reading it from incorrect place, it was working before? or it was not working from beginning? check other sensors

**Comment by Kepro** [comment]
then again, check other sensors and use sensor that have valid data

**Comment by superoloyt** [comment]
Other sensor have valid data but I would like to know the grid consumption and resfeatures too.

**Comment by superoloyt** [comment]
Other sensor have valid data but I would like to know the grid return and consumption too.

**Comment by Kepro** [comment]
show screenshot from foxcloud or mobile app if you see that there

**Comment by superoloyt** [comment]
![IMG_20230810_150311](https://github.com/macxq/foxess-ha/assets/69038702/cc42c7cc-dc19-4972-b186-4c47e00e3edf)

**Comment by Kepro** [comment]
and you don't have same value in HA? on some sensor?

**Comment by superoloyt** [comment]
Other sensors show correct value but this one gives error like on the image above.

**Comment by Lat91** [comment]
Same issue here. It's been working fine for months but upgraded ha core and this plug-in and now I have this issue. Will try and debug it soon unless anyone has the answer ?

**Comment by Kepro** [comment]
![image](https://github.com/macxq/foxess-ha/assets/1714370/d017388f-134f-4056-8c14-72e2a3d0991e)
just use feedin power and grid consumption power

**Comment by Lat91** [comment]
Not a bug in the plug-in; I had to contact my installer as the Fox maintenance on 7th August had stopped my cat clamp from working. They remotely re-enabled it and voila, I have data on those sensors again.

**Comment by superoloyt** [comment]
> ![image](https://user-images.githubusercontent.com/1714370/260229049-d017388f-134f-4056-8c14-72e2a3d0991e.png) just use feedin power and grid consumption power

How do I do that? It doesn't let me select it.

**Comment by pkasarda** [comment]
can you share guys what sensors do you see in your HA?

**Comment by kamyk** [comment]
Hello,
I have the same issue. I can not choose anything what gives me a correct value in that place:

<img width="391" alt="Screenshot 2023-10-12 at 13 57 14" src="https://github.com/macxq/foxess-ha/assets/1186538/86b6e4bb-323a-4fb3-a54c-4307cc1240df">

FoxESS - FeedIn give me correct value in entity dashboard but when I set it up I see that error too:

<img width="392" alt="Screenshot 2023-10-12 at 13 58 46" src="https://github.com/macxq/foxess-ha/assets/1186538/36debda3-882a-4555-a906-d5ce0855ede3">

Thank you for your help!

**Comment by laura1050** [comment]
I had the same problem whereby feedin and grid consumption sensors from the foxess integration have always shown as unknown.  I assumed it was because I have an old grid tied inverter (F3600).  I had to create a rieman sum integral helper to calculate the value of feedin in kwH based on input sensor
sensor.foxess_feedin_power.  That gives me a value in kwH that I added to the energy stats page.  Unfortunately I don't know how to get the yaml code for the helper to show you

**Comment by talalhassan01** [comment]
I'm also facing the same issue. All "power" sensors are not showing up in the drop downs on the Energy Dashboard. Any help on this would be appreciated.
TIA!

**Comment by talalhassan01** [comment]
Is it really this simple?

![image](https://github.com/user-attachments/assets/589d9499-beee-45f7-9b35-d6bee1732efa)
![image](https://github.com/user-attachments/assets/aae3c622-5091-4af2-b54c-133a54aa2634)
![image](https://github.com/user-attachments/assets/ec2ce7ad-bed2-4011-9c10-107bab2d04b7)

**Comment by Daibutt** [comment]
Yes! Energy dashboard displays energy (kWh) not power (kW). 
The Riemann sum templates integrate power readings into energy totals.
